src/index.js

```diff
@@ -1,81 +1,99 @@
-import React, { useRef, useState, useEffect } from 'react'
-import PropTypes from 'prop-types'
+import React, { useRef, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 
-const Collapsable = ({
+const Collapsible = ({
+  isOpen,
+  minAnimationDuration,
+  maxAnimationDuration,
+  speedDivider,
+  easing,
+  children,
+  keepChildrenOnClose,
+  ...restProps
+}) => {
+  const content = useRef();
+  const [state, setState] = useState({
+    height: isOpen ? 'auto' : 0,
+    speed: 0,
+  });
+
+  const [effectiveChildren, setChildren] = useState(children);
```

Comment:
Storing the children in state seems a bit over the top, could we not just add another option to state like `showChildren: boolean` and combine with `keepChildrenOnClose`  to conditionally add the children in the JSX?

Reply:
Also, adding / removing children when opening / closing is causing more work for react as it's having to add / remove them from the DOM, so do you have any evidence of performance gains?

Reply:
> Storing the children in state seems a bit over the top, could we not just add another option to state like showChildren: boolean and combine with keepChildrenOnClose to conditionally add the children in the JSX?

I guess adding `showChildren` will be just an extra prop. As most of the time lib user will pass isOpen value on showChildren. Also, it can't be handled outside as the effectiveChildren is set to null only after transition is finished. so then you will have to pass an onTransitionComplete prop, or onOpen/onClose prop to parent.

> Also, adding / removing children when opening / closing is causing more work for react as it's having to add / remove them from the DOM, so do you have any evidence of performance gains?

So, this comes down to this question.
-> Do you want to pretender everything so when a user performs an actions the transition is quick.
-> Do you want to delegate the work when the user want to perform an action.

On the first approach, you are paying the cost of rendering upfront for some content user may never see, probably on page load.
Plus as other things are going during page rendering, you are also adding some of the rendering time for an offscreen content.
This is valid on the update as well, why we need to do reconciliation (VDOM diffing) on offscreen content.

And on the second approach as onClick, you are doing relatively less work by shifting your work to onClick which will not be noticed. I feel this is a valid tradeoff to improve onscreen content performance. Sometime you may need a little bit of tinkering, but that you can do with `keepChildrenOnClose` prop.

Reply:
My main thoughts on this is accessibility and not hiding content from screen readers. Having a prop puts this in the hands of the developer, but I think having the children when closed should be the default.

Reply:
As for storing children in state, we can simply use a value stored in state to conditionally render the children and isOpen will override it. 

I think storing children in state and removing them constantly would be inefficient and there is something we can do with a simple Boolean instead.

Reply:
> My main thoughts on this is accessibility and not hiding content from screen readers. Having a prop puts this in the hands of the developer, but I think having the children when closed should be the default.

`display: none` and `visibility: hidden` elements are not accessible on-screen reader as well. In Ideal situation content wise we see should be similar to what screen reader reads. Providing them two versions is anti-pattern for accessibility.  Exceptions are visual contents like img, icons etc. But textual content-wise it should be same for screen reader and normal user.

Reply:
> I think storing children in state and removing them constantly would be inefficient and there is something we can do with a simple Boolean instead.

Will make it boolean based. Removing and adding will not have any perf impact over boolean as we are just keeping the reference. 
But one wrong thing I can think of here is when in the closed state the children reference might get stale in the state. 

I will update it boolean based.